### PR TITLE
conditionally render buttons based on collection visibility and access_type

### PR DESCRIPTION
**Pull request by Bento007**
#### Reviewers
**Functional:** @tihuan 

**Readability:** @seve 

---

## Changes
- modify collections page to hide delete and dataset error details.
- View of a **private** dataset as an **owner**. Errors and delete are visible.
Screen Shot 2021-02-16 at 10 52 28 AM

- View of a **published** dataset as an **owner**. Delete is not visible, and errors would be visible
Screen Shot 2021-02-16 at 11 35 52 AM

- View of a **published** dataset as a **non-owner**. Delete and error are not visible.
Screen Shot 2021-02-16 at 11 13 28 AM

### CODE REVIEW

**Review by seve — APPROVED** [review]

File: frontend/src/common/entities.ts
Comment (seve) [inline frontend/src/common/entities.ts:84]: Is collection_access_type a field on dataset? I don't see it on the [corpora-api.yaml](https://github.com/chanzuckerberg/corpora-data-portal/blob/8ad48e003283b328d7bd75ffc44140a6400479b0/backend/config/corpora-api.yml#L533)

Comment (Bento007) [reply frontend/src/common/entities.ts:84]: it's a field on collections. Does it also need to be a field on dataset?

Comment (Bento007) [reply frontend/src/common/entities.ts:84]: I was pulling this value from the collection this dataset is associated with.

**Comment by tihuan** [comment]
Thanks so much for the PR, Trent!

Given that we don't allow a private collection with invalid/failed dataset to be published, we can remove the logic for hiding the error message based on write access, I believe

The following PR shows the conditions that need to be met in order to publish a private collection:

https://github.com/chanzuckerberg/corpora-data-portal/pull/893

Hope this makes sense! Thank you!

File: frontend/src/common/entities.ts
Comment (tihuan) [reply frontend/src/common/entities.ts:84]: Ah since the property is on the collection, we don't need to add `collection_access_type` on dataset object, since that implies `dataset.collection_access_type` is available, which it's not!

File: frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/index.tsx
Comment (tihuan) [inline frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/index.tsx:159]: I think we can remove this change given that conditions needed to publish a private collection

File: frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/index.tsx
Comment (tihuan) [inline frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/index.tsx:163]: This can be removed too

File: frontend/src/views/Collection/index.tsx
Comment (tihuan) [inline frontend/src/views/Collection/index.tsx:102]: Feels like this is the only change we need for this PR!

**Review by tihuan — APPROVED** [review]
LGTM after keeping only the delete button change 🙏 Thanks so much for fixing this!

File: frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/index.tsx
Comment (Bento007) [reply frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/index.tsx:159]: in the future when we share a private collection will we want users to see the errors?

File: frontend/src/common/entities.ts
Comment (Bento007) [inline frontend/src/common/entities.ts:84]:
```suggestion
```

**Pull request merged**
